### ISSUE TITLE
"thickness" of 2D images

### ISSUE THREAD

**Issue by martinschorb** [mobie/mobie-viewer-fiji#1055]
Hi,

when I display 2D images together with volumes, it is necessary to assign a "thickness" to them. At the moment, this displayed "thickness" is determined by the source metadata ("pixel size" in *z* if present).

This is very inconsistent:

- it is not clear what is the "thickness" that the MoBIE project Creator assigns (I assume it is from the image metadata), neither how this is handled in `mobie-python` and if these are consistent.
- for `BDV-XML` formats this metadata is required (3D is a must) while in OME-Zarr, the *z* dimension is optional (if I understand it correctly).
- while the XMLs are usually part of the repository, OME-Zarr metadata is usually not and most of the time even resides on S3. And on top of it it contains pixel size information for each downsampling level. Sometimes the downsampling modifies the virtual *z* pixel size, sometimes it doesn't.
- When I now want to change the displayed "thickness" in a view, I need to manually dig out the virtual *z* pixel size and juggle around with scaling transformations for each source. :confused:  
- 
- Alternatively, the image metadata would need to be modified. As stated above, this can be very messy (especially  for multi-scale OME-Zarr).

In conclusion, I suggest that MoBIE ignores the *z* pixel size for 2D data and introduces a new field in the source definition that dettermines the slice thickness of that 2D source.

**Comment by tischi** [comment]
Overall I agree with you.

See also "Story 9" here: https://github.com/ome/ngff/issues/84

> When I now want to change the displayed "thickness" in a view, I need to manually dig out the virtual z pixel size and juggle around with scaling transformations for each source. 

But maybe that's just how it is ☝️ ?

----

I would have to dig into the Java code how this is currently handled.

I am not sure how easy it will be for me to change the voxel size metadata of the loaded image, which I _think_ would need to do in order to overwrite it with the "thickness" given in the JSON.
